vortex-layout/src/layouts/filter.rs

```diff
@@ -106,25 +107,44 @@ impl LayoutReader for FilterLayoutReader {
 
         // Otherwise, we create a new evaluation of the filter expression for this particular
         // row range.
-        let conjunct_evals: Vec<_> = filter_expr
+        let _conjunct_evals: Vec<_> = filter_expr
             .conjuncts
             .iter()
             .map(|expr| self.child.filter_evaluation(row_range, expr))
             .try_collect()?;
 
-        Ok(Box::new(FilterEvaluation {
-            filter_expr,
-            conjunct_evals,
-        }))
+        unimplemented!("FilterLayoutReader::filter_evaluation is deprecated");
     }
 
     fn projection_evaluation(
         &self,
         row_range: &Range<u64>,
         expr: &ExprRef,
     ) -> VortexResult<Box<dyn ArrayEvaluation>> {
-        // Pass-through all projection expressions to the child layout reader.
-        self.child.projection_evaluation(row_range, expr)
+        // If the expression resolves to a boolean, we split it into conjuncts for evaluation.
```

Comment:
This isn't quite right, but it's enough to test for a performance regression.

We might actually need to maintain a boolean array instead of a mask, and then check for true_count to bail early. i.e. "not false and not null", in order to maintain the correct null semantics of the expression.

Reply:
fwiw this is the error
```
FAILED test/test_dataset.py::test_to_table - pyo3_runtime.PanicException: Masks must have the same length
Backtrace:
   0: vortex_mask::bitops::<impl core::ops::bit::BitAnd for &vortex_mask::Mask>::bitand
   1: <vortex_layout::layouts::filter::FilterEvaluation as vortex_layout::reader::ArrayEvaluation>::invoke::{{closure}}
   2: vortex_layout::scan::tasks::split_exec::{{closure}}
   3: vortex_layout::scan::tasks::split_exec::{{closure}}
   4: <vortex_array::stream::adapter::ArrayStreamAdapter<S> as futures_core::stream::Stream>::poll_next
```